lib/src/components/NxCollapsibleItemsSelect/NxCollapsibleRadioSelect/__tests__/NxCollapsibleRadioSelect.test.tsx

```diff
@@ -4,173 +4,495 @@
+          });
+
+      it('has a specified classname when the tooltipModifierClass prop is provided', async function() {
+        const user = userEvent.setup(),
+            trigger = renderAndGetTriggerWrapper({
+              disabled: true,
+              disabledTooltip: 'tip',
+              tooltipModifierClass: 'customClass'
+            })!;
+
+        await runTimers();
+        expect(screen.queryByRole('tooltip')).not.toBeInTheDocument();
+
+        await user.hover(trigger);
+        await runTimers();
+
+        const tooltip = screen.getByRole('tooltip');
+        expect(tooltip.querySelector('.customClass')).toBeInTheDocument();
+      });
     });
+  });
+
+  it('calls onChange with options\'s id when an option is toggled', async function() {
+    const user = userEvent.setup(),
+        onChange = jest.fn(),
+        view = quickRender({ onChange }),
+        radios = view.getAllByRole('menuitemradio');
+
+    expect(onChange).not.toHaveBeenCalled();
+
+    await user.click(radios[0]);
+    expect(onChange).toHaveBeenCalledTimes(1);
+    expect(onChange).toHaveBeenCalledWith('foo');
+
+    await user.click(radios[1]);
+    expect(onChange).toHaveBeenCalledTimes(2);
+    expect(onChange).toHaveBeenCalledWith('bar');
+
+    await user.click(radios[2]);
+    expect(onChange).toHaveBeenCalledTimes(3);
+    expect(onChange).toHaveBeenCalledWith(null);
+  });
+
+  it('doesn\'t call onChange when options are disabled', async function() {
+    const user = userEvent.setup(),
+        onChange = jest.fn(),
+        view = quickRender({ onChange, disabled: true }),
+        radios = view.getAllByRole('menuitemradio');
 
-    it('renders unchecked <NxRadio> for all options if selectedId prop is not provided', function () {
-      const shallowRender = getShallow();
-      const renderOption = shallowRender.renderProp('renderOption');
+    expect(onChange).not.toHaveBeenCalled();
+    expect(radios[0]).toBeDisabled();
 
-      expect(renderOption({id: 'foo', name: 'Foo'})).toHaveProp('isChecked', false);
-      expect(renderOption({id: 'bar', name: 'Bar'})).toHaveProp('isChecked', false);
-      expect(renderOption({id: null, name: 'Null'})).toHaveProp('isChecked', false);
+    await user.click(radios[0]);
+    expect(onChange).not.toHaveBeenCalled();
+  });
+
+  describe('collapsible items', function() {
+    it('renders options with role=menuitemradio in an element with role=menu', function() {
+      const view = quickRender(),
+          menu = view.getByRole('menu'),
+          options = within(menu).getAllByRole('menuitemradio');
+
+      expect(menu).toBeInTheDocument();
+      expect(options.length).toBe(3);
     });
 
-    it('renders enabled <NxRadio> for all options if component is not disabled', function () {
-      const renderOption = getShallow().renderProp('renderOption');
+    it('renders text content according to option\'s name prop', function() {
+      const options = quickRender().getByRole('menu').children;
```

Comment:
This is assuming an implementation detail a little bit.  This test can either just be removed, or you can adjust it to get all of the `menuitemradio`s and then get their first label elements (e.g. `expect(radios[0].label[0]).toHaveTextContent('Foo')`)

Reply:
> This is assuming an implementation detail a little bit. This test can either just be removed, or you can adjust it to get all of the `menuitemradio`s and then get their first label elements (e.g. `expect(radios[0].labels[0]).toHaveTextContent('Foo')`)

Since we are trying to steer away from implementation details, I just removed the test here: [f1e7c2f](https://github.com/sonatype/sonatype-react-shared-components/pull/932/commits/f1e7c2f7c0b6bdc41fa1a9be7801faba53a9503d)